File: apps/backend/app/prompts/enrichment.py
"""LLM prompt templates for AI-powered resume enrichment."""

ANALYZE_RESUME_PROMPT = """You are a professional resume analyst. Analyze this resume to identify items in Experience and Projects sections that have weak, vague, or incomplete descriptions.

IMPORTANT: Generate ALL output text (questions, placeholders, summaries, weakness reasons) in {output_language}.

RESUME DATA (JSON):
{resume_json}

WEAK DESCRIPTION INDICATORS:
1. Generic phrases: "responsible for", "worked on", "helped with", "assisted in", "involved in"
2. Missing metrics/impact: No numbers, percentages, dollar amounts, or measurable outcomes
3. Unclear scope: Vague about team size, project scale, user count, or responsibilities
4. No technologies/tools: Missing specific tech stack, tools, or methodologies used
5. Passive voice without ownership: Not clear what the candidate personally accomplished
6. Too brief: Single short bullet that doesn't explain the work

GOOD DESCRIPTION EXAMPLES (for reference):
- "Led migration of 15 microservices to Kubernetes, reducing deployment time by 60%"
- "Built real-time analytics dashboard using React and D3.js, serving 10K daily users"
- "Architected payment processing system handling $2M monthly transactions"

TASK:
1. Review each Experience and Project item's description bullets
2. Identify items that would benefit from more detail
3. Generate a MAXIMUM of 6 questions total across ALL items (not per item)
4. Prioritize the most impactful questions that will yield the best improvements
5. If multiple items need enhancement, distribute questions wisely (e.g., 2-3 per item)
6. Questions should help extract: metrics, technologies, scope, impact, and specific contributions

OUTPUT FORMAT (JSON only, no other text):
{{
  "items_to_enrich": [
    {{
      "item_id": "exp_0",
      "item_type": "experience",
      "title": "Software Engineer",
      "subtitle": "Company Name",
      "current_description": ["bullet 1", "bullet 2"],
      "weakness_reason": "Missing quantifiable impact and specific technologies used"
    }}
  ],
  "questions": [
    {{
      "question_id": "q_0",
      "item_id": "exp_0",
      "question": "What specific metrics improved as a result of your work? (e.g., performance gains, cost savings, user growth)",
      "placeholder": "e.g., Reduced API response time by 40%, saved $50K annually"
    }},
    {{
      "question_id": "q_1",
      "item_id": "exp_0",
      "question": "What technologies, frameworks, or tools did you use in this role?",
      "placeholder": "e.g., Python, FastAPI, PostgreSQL, Redis, AWS Lambda"
    }},
    {{
      "question_id": "q_2",
      "item_id": "exp_0",
      "question": "What was the scale of your work? (team size, users served, data volume)",
      "placeholder": "e.g., Team of 5, serving 100K users, processing 1M requests/day"
    }},
    {{
      "question_id": "q_3",
      "item_id": "exp_0",
      "question": "What was your specific contribution or ownership in this project?",
      "placeholder": "e.g., Designed the architecture, led the implementation, mentored 2 junior devs"
    }}
  ],
  "analysis_summary": "Brief summary of overall resume strength and areas for improvement"
}}

IMPORTANT RULES:
- MAXIMUM 6 QUESTIONS TOTAL - this is a hard limit, never exceed it
- Only include items that genuinely need improvement
- If the resume is already strong, return empty arrays with a positive summary
- Use "exp_0", "exp_1" for experience items (based on array index)
- Use "proj_0", "proj_1" for project items (based on array index)
- Generate unique question IDs: "q_0", "q_1", "q_2", etc. (max q_5)
- Questions should be specific to the role/project context
- Keep questions conversational but professional
- Placeholder text should give concrete examples
- Prioritize quality over quantity - ask the most impactful questions first"""

ENHANCE_DESCRIPTION_PROMPT = """You are a professional resume writer. Your goal is to ADD new bullet points to this resume item using the additional context provided by the candidate. DO NOT rewrite or replace existing bullets - only add new ones.

IMPORTANT: Generate ALL output text (bullet points) in {output_language}.

ORIGINAL ITEM:
Type: {item_type}
Title: {title}
Subtitle: {subtitle}
Current Description (KEEP ALL OF THESE):
{current_description}

CANDIDATE'S ADDITIONAL CONTEXT:
{answers}

TASK:
Generate NEW bullet points to ADD to the existing description. The original bullets will be kept as-is.
New bullets should be:
1. Action-oriented: Start with strong verbs (Led, Built, Architected, Implemented, Optimized)
2. Quantified: Include metrics, numbers, percentages where the candidate provided them
3. Technically specific: Mention technologies, tools, and methodologies
4. Impact-focused: Clearly state the business or technical outcome
5. Ownership-clear: Show what the candidate personally did vs. the team

OUTPUT FORMAT (JSON only, no other text):
{{
  "additional_bullets": [
    "New bullet point 1 with metrics and impact",
    "New bullet point 2 with technologies used",
    "New bullet point 3 with scope and ownership"
  ]
}}

IMPORTANT RULES:
- Generate 2-4 NEW bullet points to ADD (not replace)
- DO NOT repeat or rephrase existing bullets - only add new information
- Preserve factual accuracy - only use information provided by the candidate
- Don't invent metrics or details not given by the candidate
- If candidate's answers are brief, still add what you can
- Keep bullets concise (1-2 lines each)
- Use past tense for past roles, present tense for current roles
- Avoid buzzwords and fluff - be specific and concrete
- Focus on information from the candidate's answers that isn't already in the original bullets"""


# ============================================
# AI Regenerate Feature Prompts
# ============================================


REGENERATE_ITEM_PROMPT = """You are a professional resume writer. Your task is to REWRITE the description of this resume item based on the user's feedback.

IMPORTANT: Generate ALL output text in {output_language}.

ITEM INFORMATION:
Type: {item_type}
Title: {title}
Subtitle: {subtitle}

CURRENT DESCRIPTION (the user is NOT satisfied with this):
{current_description}

USER'S FEEDBACK/INSTRUCTION:
{user_instruction}

TASK:
Based on the user's feedback, completely REWRITE the description bullets. The new description should:
1. Address the user's specific concerns/requests
2. Be action-oriented with strong verbs
3. Highlight quantifiable impact ONLY when it already exists in the current description or the user's feedback (never invent numbers)
4. Be technically specific with tools/technologies
5. Show clear impact and ownership

OUTPUT FORMAT (JSON only):
{{
  "new_bullets": [
    "Completely rewritten bullet point 1",
    "Completely rewritten bullet point 2",
    "Completely rewritten bullet point 3"
  ],
  "change_summary": "Brief explanation of what was changed based on user feedback"
}}

RULES:
- Generate 2-5 NEW bullets (not additions, but replacements)
- Directly address the user's instruction
- Do NOT add any new facts, metrics, dates, companies, titles, or accomplishments that are not already present in CURRENT DESCRIPTION or USER'S FEEDBACK/INSTRUCTION
- If the user asks for metrics but none exist in the provided text, do not fabricate numbers; rewrite to emphasize scope/impact qualitatively instead
- Keep bullets concise (1-2 lines each)
- Use past tense for past roles, present tense for current"""


REGENERATE_SKILLS_PROMPT = """You are a professional resume writer. Reorganize and regroup the candidate's technical skills based on user feedback.

IMPORTANT: Generate ALL output text in {output_language}.

MASTER PROFILE SKILLS (source of truth — only use skills from this list):
{master_skills}

CURRENT TAILORED SKILLS:
{current_skills}

USER'S FEEDBACK:
{user_instruction}

OUTPUT FORMAT (JSON only):
{{
  "skillGroups": [
    {{"name": "Languages", "skills": ["Python", "TypeScript"]}},
    {{"name": "Frontend", "skills": ["React", "Next.js"]}},
    {{"name": "Backend", "skills": ["FastAPI", "Node.js"]}},
    {{"name": "Databases", "skills": ["PostgreSQL", "Redis"]}},
    {{"name": "Cloud & DevOps", "skills": ["AWS", "Docker"]}},
    {{"name": "Tools", "skills": ["Git", "Linux"]}}
  ],
  "change_summary": "Brief explanation of changes"
}}

CATEGORY GUIDELINES:
- Languages: Programming languages (Python, JavaScript, TypeScript, etc.)
- Frontend: Frontend frameworks, libraries, and UI tools (React, Vue, Tailwind, etc.)
- Backend: Backend frameworks and API technologies (FastAPI, Express, GraphQL, etc.)
- Databases: Databases and ORMs (PostgreSQL, MongoDB, Prisma, etc.)
- Cloud & DevOps: Cloud platforms, containers, CI/CD (AWS, Docker, GitHub Actions, etc.)
- Architecture: Design patterns, system design concepts (Microservices, SOLID, etc.)
- AI/LLM: AI/ML frameworks and concepts (LangChain, PyTorch, RAG, etc.)
- Tools: Dev tools, version control, testing, productivity (Git, Jest, Figma, etc.)

RULES:
- ONLY include skills that exist in MASTER PROFILE SKILLS — never invent or add new skills
- Group each skill into the most appropriate category
- A skill should appear in only one category
- If a category would be empty, omit it entirely
- Prioritize and reorder skills within each group based on the user's feedback
- Keep skill names exactly as they appear in MASTER PROFILE SKILLS (do not rename)
- 2-3 categories is fine if the candidate has a focused skill set"""


REGENERATE_SUMMARY_PROMPT = """You are an expert technical recruiter, ATS resume strategist, and professional resume writer.

IMPORTANT: Generate ALL output text in {output_language}.

Your task is to create a high-impact, ATS-friendly professional summary from the candidate's resume.

Do NOT immediately write the summary. Follow the analysis process below first.

STEP 1 — Understand the Candidate

Analyze the resume and identify:
- Current/professional identity
- Target career direction
- Technical domain
- Most relevant technologies and skills
- Strongest projects
- Most relevant work/internship experience
- Most impressive measurable achievements
- Business or technical impact
- Areas of specialization
- Level of experience
- Education only when it strengthens the positioning

Separate facts explicitly supported by the resume from assumptions.
Never invent: experience, years of experience, technologies, job titles, metrics, responsibilities, achievements, or certifications.
If a metric is not available, do not fabricate one.

STEP 2 — Analyze the User's Instruction

Read the user's feedback carefully. Determine what they want changed:
- Tone (more formal, more casual, more confident)
- Focus (emphasize certain skills, projects, or achievements)
- Length (shorter, longer)
- Style (more concise, more detailed, different structure)
- Target role or industry alignment

The final summary MUST address the user's specific request.

STEP 3 — Select the Strongest Evidence

Choose the 2-3 strongest pieces of evidence from the resume. Prioritize:
1. Quantified achievements
2. Relevant production/project experience
3. Relevant technical expertise
4. Scale or complexity
5. Business impact
6. Leadership/ownership when relevant

Prefer: "Built X that improved Y by Z%" over "Responsible for building X."

STEP 4 — Determine the Candidate's Positioning

Answer internally: "Why should a recruiter consider this candidate?"
Create a one-sentence positioning statement.

STEP 5 — Write the Summary

Write a 2-4 sentence professional summary:
- Sentence 1 (Identity): professional identity and career direction
- Sentence 2 (Expertise): 2-4 most relevant technical/domain strengths
- Sentence 3 (Evidence): strongest relevant achievement(s), preferably quantified
- Sentence 4 (Value): type of value the candidate brings

Do not force all four sentences if a shorter summary is stronger.
The summary MUST reflect the user's instruction from STEP 2.

STEP 6 — Optimize for ATS

- Use exact technology names when truthful
- Avoid keyword stuffing and unnecessary synonyms
- Keep the language simple and machine-readable

STEP 7 — Make It Human

Avoid generic phrases: "passionate professional", "results-driven", "highly motivated", "dynamic individual", "team player", "proven track record" (unless supported by evidence), "responsible for", "seeking to leverage my skills".
Replace vague claims with evidence. Use clear, confident, natural language.

STEP 8 — Final Quality Check

Verify: relevance, evidence support, ATS readability, human readability, accuracy (no invented facts/metrics).

OUTPUT FORMAT (JSON only):

{{
  "candidate_positioning": "One sentence explaining how the candidate should be positioned",
  "key_evidence": ["Evidence 1", "Evidence 2", "Evidence 3"],
  "new_summary": "The final polished 2-4 sentence professional summary",
  "alternative_summary": "A second slightly more human and recruiter-friendly version",
  "recruiter_score": 85,
  "score_explanation": "Brief explanation of the score",
  "change_summary": "Brief explanation of what was changed"
}}

IMPORTANT RULES:
1. Never fabricate information, metrics, technologies, job titles, or achievements
2. Never claim expertise the resume does not demonstrate
3. Prioritize evidence over adjectives, achievements over responsibilities
4. Keep the final summary concise (2-4 sentences)
5. Do not mention weaknesses or missing requirements
6. Do not use first person ("I", "my", "me")
7. Do not mention years of experience unless explicitly provided
8. The summary must be optimized for both ATS parsing and human recruiters
9. The summary MUST be different from the original if the user requested changes
10. Return ONLY the JSON object, no other text

## INPUT

### RESUME

{resume}

### USER'S INSTRUCTION

{user_instruction}"""


# ============================================
# Project Bullet Generation
# ============================================


GENERATE_PROJECT_BULLETS_PROMPT = """You are a professional resume writer. Write resume bullet points for a personal project section.

IMPORTANT: Generate ALL output text in {output_language}.

PROJECT INFORMATION:
Name: {name}
Role: {role}
Years: {years}
GitHub: {github}
Website: {website}
Description: {description}
Languages: {languages}

PROJECT SOURCE MATERIAL (the primary source of facts about this project):
{source_material}

{user_instruction}
TASK:
Write bullet points for this project's resume entry. The bullets should:
1. Be action-oriented with strong verbs
2. Highlight quantifiable impact ONLY when the source material supports it (never invent numbers)
3. Be technically specific with the tools, technologies, and architecture described
4. Show ownership, scope, and what was built or delivered
5. Be concise (1-2 lines each)

OUTPUT FORMAT (JSON only):
{{
  "bullets": [
    "Bullet point 1",
    "Bullet point 2",
    "Bullet point 3"
  ]
}}

RULES:
- Generate 3-5 bullets
- Base every bullet ONLY on the PROJECT SOURCE MATERIAL and PROJECT INFORMATION above
- Do NOT invent features, metrics, or outcomes that are not present in the source material
- Do NOT mention or reference the source document (e.g. "README")
- If the source material is thin, write honest bullets about the project scope, technologies, and role without fabricating details"""


# ============================================
# Outreach Email Generation Prompt
# ============================================


GENERATE_OUTREACH_EMAIL_PROMPT = """You are an experienced outreach writer. Write a short, highly personalized email to a company based on the candidate's profile and the target role.

IMPORTANT: Generate ALL output text in {output_language}.

### Inputs

**Candidate Profile:**
{sender_info}

**Company Information:**
- Company name: {company_name}
- Company email: {company_email}
- Industry: {industry}
- Company size: {company_size}
- Company type: {company_type}
- Website: {website}
- LinkedIn: {linkedin_url}
- Recipient name (if known): {recipient_name}

**Purpose:**
{purpose}

---

### Main Goal

Write an email that feels like it was **personally written by a real candidate**, not generated by AI.

The recipient should quickly understand:
1. Who the candidate is
2. Why they are contacting this company
3. What relevant experience or value they bring
4. What they are asking for

The email should make the reader think: "This person actually looked at our company and has a relevant background."

---

### Humanization Rules

**1. Sound like a real person.**
Use natural, conversational professional language. Avoid overly polished corporate language such as:
- "I am thrilled to express my interest..."
- "I believe I would be an exceptional fit..."
- "I am reaching out to explore potential synergies..."
- "Your esteemed company..."
- "I am passionate about contributing to your innovative organization..."

Prefer simple language:
- "I came across your company and..."
- "The role caught my attention because..."
- "I've been working on..."
- "I thought I'd reach out because..."

**2. Do NOT sound like AI.**
Avoid: generic compliments, repeating the job description, excessive adjectives, buzzword-heavy sentences, perfectly symmetrical paragraphs, artificial enthusiasm, empty statements about "passion" or "innovation."

Do not try to make every sentence impressive. **Natural and specific is better than impressive and generic.**

**3. Personalize the email.**
Mention 1-2 concrete details about the company (from the information above). Connect those to something genuinely relevant in the candidate's background. Never invent experience.

**4. Keep it short: 80-140 words.** Every sentence should have a purpose. Do not turn the email into a cover letter.

**5. Make the opening interesting.** Start directly — do NOT open with "I hope you're doing well" or "I am writing to express my interest." Instead: "I came across [company] and noticed you're working with [tech/industry]."

**6. Focus on relevance, not self-promotion.** Select the 2-3 strongest pieces of evidence that make the candidate relevant to this particular company. Use concrete outcomes whenever available.

**7. PROJECT MATCHING:** Look at the sender's Projects list. Pick 1-2 projects whose technologies or purpose most closely match what the company does. Mention these projects briefly by name and what they demonstrate. Do NOT list all projects. When linking to a project, use the **Live** URL (website) if available — only use the **GitHub** URL if the project has no website.

**8. Avoid desperation.** Never write "I desperately need an opportunity" or "Please consider my application." The tone should be confident, interested, and professional.

**9. Don't overuse first person.** Avoid starting every sentence with "I have... I worked on... I developed..."

**10. Preserve authenticity.** If the candidate is junior, do not artificially make them sound senior. Authenticity is more important than sounding impressive.

---

### Output Format (JSON only):
{{
  "subject": "A concise, attention-grabbing subject line",
  "body": "The full email body in markdown format"
}}

### Subject Line Rules
- Keep it between 4-9 words whenever possible.
- Make it specific to the role or company.
- Include the candidate's name only when useful.
- Mention a relevant skill or background when it adds value.
- Avoid generic subjects like "Job Application", "Looking for Opportunities", "Interested in Your Company".
- Avoid excessive enthusiasm like "Exciting Opportunity!" or "Dream Job!".
- No emojis, no ALL CAPS, no sales-sounding language.
- Prioritize specificity over creativity. **Clarity > creativity.**
- Generate 3 possible subject lines internally, then select the best one that is most relevant, most natural, and shortest.

Good examples:
- Backend Developer — Interested in the [Role] position
- Backend Developer | Node.js & AWS
- Reaching out about your Backend Engineer role
- Junior Software Engineer — [Candidate Name]
- [Role] opportunity at [Company]

### Formatting Rules:
- Use markdown. **Bold** for emphasis. [text](url) for links.
- Blank lines between paragraphs. No bullet points — flowing paragraphs only.
- When linking projects: use **Live** URL first, **GitHub** URL as fallback.
- Sign with the candidate's name if available, otherwise "Sincerely".
- Do NOT include a contact footer — it is appended automatically.
- Do NOT invent facts about the company or candidate beyond what is provided.
- Do NOT use placeholders like [Your Name].

### Final Quality Check (silently):
- Does this sound like a real person wrote it?
- Did I mention something specific about this company?
- Did I connect that to actual candidate experience?
- Is it under 140 words?
- Does the CTA feel natural rather than desperate?

Return ONLY the JSON. No explanation."""

File: apps/frontend/lib/api/enrichment.ts
/**
 * API functions for AI-powered resume enrichment.
 */

import { apiFetch, apiPost } from './client';

// Types matching backend schemas

export interface EnrichmentItem {
  item_id: string;
  item_type: 'experience' | 'project';
  title: string;
  subtitle?: string;
  current_description: string[];
  weakness_reason: string;
}

export interface EnrichmentQuestion {
  question_id: string;
  item_id: string;
  question: string;
  placeholder: string;
}

export interface AnalysisResponse {
  items_to_enrich: EnrichmentItem[];
  questions: EnrichmentQuestion[];
  analysis_summary?: string;
}

export interface AnswerInput {
  question_id: string;
  answer: string;
}

export interface EnhancedDescription {
  item_id: string;
  item_type: 'experience' | 'project';
  title: string;
  original_description: string[];
  enhanced_description: string[];
}

export interface EnhancementPreview {
  enhancements: EnhancedDescription[];
}

/**
 * Analyze a resume to identify items that need enrichment.
 * Returns items with weak descriptions and clarifying questions.
 */
export async function analyzeResume(resumeId: string): Promise<AnalysisResponse> {
  const res = await apiFetch(`/enrichment/analyze/${resumeId}`, {
    method: 'POST',
    credentials: 'include',
  });

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to analyze resume (status ${res.status}).`);
  }

  return res.json();
}

/**
 * Generate enhanced descriptions from user answers.
 */
export async function generateEnhancements(
  resumeId: string,
  answers: AnswerInput[]
): Promise<EnhancementPreview> {
  const res = await apiPost('/enrichment/enhance', {
    resume_id: resumeId,
    answers,
  });

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to generate enhancements (status ${res.status}).`);
  }

  return res.json();
}

/**
 * Apply enhancements to the master resume.
 */
export async function applyEnhancements(
  resumeId: string,
  enhancements: EnhancedDescription[]
): Promise<{ message: string; updated_items: number }> {
  const res = await apiPost(`/enrichment/apply/${resumeId}`, {
    enhancements,
  });

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to apply enhancements (status ${res.status}).`);
  }

  return res.json();
}

// ============================================
// AI Regenerate Feature Types
// ============================================

export interface RegenerateItemInput {
  item_id: string;
  item_type: 'experience' | 'project' | 'skills' | 'summary';
  title: string;
  subtitle?: string;
  current_content: string[];
}

export interface RegenerateRequest {
  resume_id: string;
  items: RegenerateItemInput[];
  instruction: string;
  output_language?: string;
}

export interface RegeneratedItem {
  item_id: string;
  item_type: 'experience' | 'project' | 'skills' | 'summary';
  title: string;
  subtitle?: string;
  original_content: string[];
  new_content: string[];
  new_skill_groups?: { name: string; skills: string[] }[];
  diff_summary: string;
}

export interface RegenerateItemError {
  item_id: string;
  item_type: 'experience' | 'project' | 'skills' | 'summary';
  title: string;
  subtitle?: string;
  message: string;
}

export interface RegenerateResponse {
  regenerated_items: RegeneratedItem[];
  errors?: RegenerateItemError[];
}

/**
 * Regenerate selected resume items based on user feedback.
 * Uses AI to rewrite content addressing user's concerns.
 */
export async function regenerateItems(request: RegenerateRequest): Promise<RegenerateResponse> {
  const res = await apiPost('/enrichment/regenerate', request);

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to regenerate content (status ${res.status}).`);
  }

  return res.json();
}

/**
 * Apply regenerated items to the master resume.
 */
export async function applyRegeneratedItems(
  resumeId: string,
  regeneratedItems: RegeneratedItem[]
): Promise<{ message: string; updated_items: number }> {
  const res = await apiPost(`/enrichment/apply-regenerated/${resumeId}`, regeneratedItems);

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to apply changes (status ${res.status}).`);
  }

  return res.json();
}

// ============================================
// Project Bullet Generation Types
// ============================================

export interface GenerateProjectBulletsRequest {
  name: string;
  role?: string | null;
  years?: string | null;
  github?: string | null;
  website?: string | null;
  description?: string[];
  languages?: string[];
  readme?: string | null;
  prompt?: string;
  output_language?: string;
}

export interface GenerateProjectBulletsResponse {
  bullets: string[];
}

/**
 * Generate resume bullet points for a project from its README/description,
 * optionally guided by a user mini-prompt.
 */
export async function generateProjectBullets(
  request: GenerateProjectBulletsRequest
): Promise<GenerateProjectBulletsResponse> {
  const res = await apiPost('/enrichment/generate-project-bullets', request);

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to generate project bullets (status ${res.status}).`);
  }

  return res.json();
}

// ============================================
// Company Research Types
// ============================================

export interface ResearchCompanyRequest {
  company_name: string;
  website?: string | null;
  industry?: string | null;
  linkedin_url?: string | null;
}

export interface ResearchCompanyResponse {
  company_name: string;
  website_content: string;
  linkedin_content: string;
  hiring_signals: string;
  research_summary: string;
  sources: string[];
}

/**
 * Research a company by crawling its website, LinkedIn page, and checking hiring signals.
 * Returns structured research data to personalize outreach emails.
 */
export async function researchCompany(
  payload: ResearchCompanyRequest
): Promise<ResearchCompanyResponse> {
  const res = await apiPost('/enrichment/research-company', payload);

  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data.detail || `Failed to research company (status ${res.status}).`);
  }

  return res.json();
}
